src/test/java/picard/TestDataProviders.java

```diff
@@ -21,21 +26,63 @@
 public class TestDataProviders {
 
     @Test
-    public void testAllDataProviders() throws IllegalAccessException, InstantiationException, InvocationTargetException {
-        int i = 0;
-        final ClassFinder classFinder = new ClassFinder();
-        classFinder.find("picard", Object.class);
-        for (Class testClass : classFinder.getClasses()) {
-            if (Modifier.isAbstract(testClass.getModifiers())) continue;
-            for (final Method method : testClass.getMethods()) {
-                if (method.isAnnotationPresent(DataProvider.class)) {
-                    System.err.println("Method: " + testClass.getName() + ":" + method.getName());
-                    method.invoke(testClass.newInstance());
-                    i++;
+    public void independentTestOfDataProviderTest() throws Exception {
+        final Iterator<Object[]> data = TestNGUtil.getDataProviders("picard");
+
+        Assert.assertTrue(data.hasNext(), "Found no data from testAllDataProvidersdata. Something is wrong.");
+
+        Assert.assertEquals( StreamSupport.stream(Spliterators.spliteratorUnknownSize(data, 0), false)
+                        .filter(c -> ((Method) c[0]).getName().equals("testAllDataProvidersdata")).count(), 1,
+                "getDataProviders didn't find testAllDataProvidersdata, which is in this class. Something is wrong.");
+    }
+
+    @DataProvider(name = "DataprovidersThatDontTestThemselves")
+    public Iterator<Object[]> testAllDataProvidersdata() throws Exception {
```

Comment:
I think capitalization should be `testAllDataProvidersData`

Reply:
done.